src/Reflector/ClassReflector.php

```diff
@@ -69,4 +79,14 @@ public function getAllClasses()
             new IdentifierType(IdentifierType::IDENTIFIER_CLASS)
         );
     }
+
+    /**
+     * Return the class context factory.
+     *
+     * @return ContextFactory
+     */
+    public function getContextFactory()
```

Comment:
The reflector holds the context factory service, and makes it available to it's offspring.